Milestone 9

pasivinay opened the pull request:
### **This PR sets up one-click deployments using ArgoCD, enabling automated deployments based on GitOps principles. ArgoCD will deploy applications automatically when code changes are merged and the CI pipeline runs successfully.**

**Changes Introduced:**

1. Deployed and configured ArgoCD in the argocd namespace on the dependent_services node.
2. Defined ArgoCD resources (apps, repository secrets, etc.) declaratively using Kubernetes manifests.
3. Committed all ArgoCD manifests to the GitHub repository with a proper file structure.
4. Configured ArgoCD to use Helm charts and values.yaml as the source of truth.
5. Implemented a GitHub Actions job to update the image tag in values.yaml and commit changes.
6. Ensured ArgoCD auto-syncs applications upon detecting new changes.
7. Updated README.md with setup instructions for ArgoCD in a Kubernetes cluster.

pasivinay committed:
Add helm charts folder in milestone_9

Added helm charts folder with application deployment charts in
milestone_9 folder.

pasivinay committed:
Add argocd helm chart

Added helm chart that deploys application with argocd.

pasivinay committed:
Add github workflow for image tag update

Added github workflow pipeline file that updated image tag dynamically.

pasivinay committed:
Remove ci-pipeline.yml pipeline file

pasivinay committed:
Rename workflow file

pasivinay committed:
Add README.md file for milestone 9

Added README.md file with instruction for successfully implementing
milestone 9.

pasivinay committed:
Modify main README.file for milestone 9

Added milestone 9 section to main README.md file.

pasivinay committed:
Update docker image tag version

Updated image tag version in github action pipeline file.

pasivinay committed:
Update tag to 1.0.0

pasivinay committed:
Update tag version to 1.0.1

pasivinay committed:
Add demo migration folder to milestone 4

Added a sample migration folder to milestone 4 folder

pasivinay committed:
Update image tag version to 2.0.0

pasivinay committed:
Merge remote-tracking branch 'origin/master' into milestone_9

pasivinay committed:
Modify application and database helm charts

Added resource constraints in application and database helm charts. Also
added resource contraint values in values.yaml file.

pasivinay committed:
Update image tag version to 1.0.1

pasivinay committed:
Fix requirements.txt file

Added pylint and PyMySql[rsa] dependencies

pasivinay committed:
Modify github action pipeline to add linting

Added linting step in github action pipeline

pasivinay committed:
Added resource limits

pasivinay committed:
Modified values files for database helm chart

Increased resource limit for database helm chart.

rajeshecb70 commented on milestone_9/helm/application/templates/application.yaml:
the selector must be parameterised

pasivinay replied:
Done

pasivinay committed:
Modify application manifest file

Parameterize the nodeselector type for the application deployment
resource with helm values file.
